### PR TITLE
[CCIP-2360] Cheaper CI Runs

### PR DESCRIPTION
**Pull request by kalverra**
## Motivation

I've done some testing on the main `chainlink` repo and was able to drop our CI costs dramatically by taking advantage of smaller runners with negligible runtime hits. [GitHub has gifted us more powerful free runners](https://github.blog/2024-01-17-github-hosted-runners-double-the-power-for-open-source/), let's use em.

## Solution

The first step was to take advantage of GitHub's more powerful free runners, so I [dropped some jobs to the free tier](https://github.com/smartcontractkit/ccip/pull/907/files#diff-4761c5069b46ce96f49d74476da61f43ced3abc73d22254cd95a2be0e250a593R304-R317).

Secondly, there were [some client compatibility tests that ran on every PR](https://github.com/smartcontractkit/ccip/pull/907/files#diff-4761c5069b46ce96f49d74476da61f43ced3abc73d22254cd95a2be0e250a593L359-L386) and used [expensive 8core runners](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#per-minute-rates-for-larger-runners). I'm removing these because:

1. We're already checking OCR and OCRv2 [compatibility on a nightly basis in the core repo](https://github.com/smartcontractkit/chainlink/blob/develop/.github/workflows/client-compatibility-tests.yml#L84). 
2. They weren't checking any compatibility on CCIP. The `client` field isn't used at all in the action. They were all just running OCR2 smoke tests on geth, but single-threaded, wasting time and money.

**My back-of-the-envelope math tells me this will save us about `10 minutes` and `$3.87` per test run. We run the tests about 10 times a day, so this will save us about `3,000 minutes` and `$1,200` a month. Even more, assuming new contributors will increase how often we run them.**

We can resurrect them in the designated CCIP compatibility tests if we deem it appropriate to. Otherwise, I've set these tests which I've [set to run nightly and on new tags](https://github.com/smartcontractkit/ccip/pull/907/files#diff-4bd01e32e25e3aed20a83e6bc207aab3e98b178e029b523e9df12ccc021a870dR3-R7) and notify us if they fail.

### CODE REVIEW

**Review by RensR — APPROVED** [review]

File: .github/workflows/ccip-client-compatibility-tests.yml [hunk 1]
```diff
@@ -1,13 +1,10 @@
...
-#  pull_request:
-#  push:
-#    tags:
-#      - "*"
-#    branches:
-#      - ccip-develop
-#      - "release/**"
+  push:
+    tags:
+      - '*'
```
Comment (AnieeG) [inline .github/workflows/ccip-client-compatibility-tests.yml:5]: want to make it for release tags only?

File: .github/workflows/ccip-client-compatibility-tests.yml [hunk 2]
```diff
@@ -411,7 +359,7 @@ jobs:
           github_repository: ${{ github.repository }}
           workflow_run_id: ${{ github.run_id }}
           github_job_name_regex: ^CCIP Latest EVM Node Compatibility Test with (?<cap>.*?)$
-          message_title: BiDirectional Lane Test Results
-          slack_channel_id: ${{ secrets.QA_SLACK_CHANNEL }}
+          message_title: CCIP Compatibility Test Results
+          slack_channel_id: "#ccip-testing"
```
Comment (AnieeG) [inline .github/workflows/ccip-client-compatibility-tests.yml]: Let's put it in another secret. wdyt?

File: .github/workflows/ccip-client-compatibility-tests.yml [hunk 3]
```diff
@@ -215,32 +166,28 @@ jobs:
...
-          # - name: bidirectionallane-besu
+          # - name: bidirectional-lane-besu
           #   test: TestSmokeCCIPForBidirectionalLane
           #   client: besu
-          #   timeout: 30m
-          #   pyroscope_env: ci-ccip-bidirectionallane-besu
+          #   pyroscope_env: ci-ccip-bidirectional-lane-besu
           #   chainConfig: "1337=hyperledger/besu:${{ needs.get-latest-available-images.outputs.besu_tag }},2337=hyperledger/besu:${{ needs.get-latest-available-images.outputs.besu_tag }}"
-          - name: bidirectionallane-erigon
+          - name: bidirectional-lane-erigon
```
Comment (AnieeG) [inline .github/workflows/ccip-client-compatibility-tests.yml]: This is gonna result into failure. Let's comment this out. There is a ticket for this.

File: .github/workflows/ccip-client-compatibility-tests.yml (hunk 1 above)
Comment (kalverra) [reply .github/workflows/ccip-client-compatibility-tests.yml:5]: My fear is that someone typos the tag name and we don't trigger, or the regex changes and no one notices. I'd rather just run it for all the tags. If that turns our to be too much, we can adjust later.

File: .github/workflows/ccip-client-compatibility-tests.yml (hunk 3 above)
Comment (kalverra) [reply .github/workflows/ccip-client-compatibility-tests.yml]: Sure. Have we tried the others though? At this point we're only testing Geth compatibility, which our other tests already do implicitly.

Comment (AnieeG) [reply .github/workflows/ccip-client-compatibility-tests.yml]: Right, others are also not tried sufficiently to prove their reliability. The scope the same ticket can be increased to enable this all later. This is the reason this workflow only supports manual triggering rn.

Comment (kalverra) [reply .github/workflows/ccip-client-compatibility-tests.yml]: Commented out and added TODO

File: .github/workflows/ccip-client-compatibility-tests.yml (hunk 2 above)
Comment (kalverra) [reply .github/workflows/ccip-client-compatibility-tests.yml]: Done

**Review by AnieeG — APPROVED** [review]

**Pull request merged**